src/main/java/org/cryptomator/integrations/keychain/KeychainAccessProvider.java

```diff
@@ -52,4 +52,18 @@
 	 */
 	boolean isLocked();
 
+	/**
+	 * @return <code>true</code> if this KeychainAccessIntegration needs to be associated with the KeychainBackend.
+	 * @implSpec This method must not throw any exceptions and should fail fast
+	 * returning <code>false</code> if the KeychainAccessIntegration was associated with the KeychainBackend and the KeychainAccessIntegration can use the KeychainBackend.
+	 */
+	default boolean needsAssociation() { return false; }
+
+	/**
+	 * @return <code>true</code> if associating the KeychainAccessIntegration with the KeychainBackend was successfull.
```

Comment:
Can you please refine the JavaDocs and specify what this method is supposed to do? I suspect this is highly KCXC-specific.